spring-integration-core/src/test/java/org/springframework/integration/message/TestHandlers.java

```diff
@@ -89,4 +90,10 @@ public final static Object countingCountDownHandler(final AtomicInteger counter,
 		};
 	}
 
+	public static class RequestHeaderCopyingEchoHandler extends AbstractReplyProducingMessageHandler {
```

Comment:
maybe extract this as a top level class or create a static method to return it? it doesn't fit the general model of the other factory methods (which return POJOs), but could simplify things.